pennylane/tape/qscript.py

```diff
@@ -428,10 +428,12 @@ def _update_par_info(self):
                 {"op": op, "op_idx": idx, "p_idx": i} for i, d in enumerate(op.data)
             )
 
+        n_ops = len(self.operations)
         for idx, m in enumerate(self.measurements):
             if m.obs is not None:
                 self._par_info.extend(
-                    {"op": m.obs, "op_idx": idx, "p_idx": i} for i, d in enumerate(m.obs.data)
+                    {"op": m.obs, "op_idx": idx + n_ops, "p_idx": i}
```

Comment:
Will mid-circuit measurements still have indices that are greater than `len(tape.operations)`?

Reply:
No. Right now, we defer mid-circuit measurements so that is not an issue. Once mid-circuit measurements are natively supported, it should still be fine since mid-circuit measurements do not currently support observables, so a mid-circuit measurement will not be present in `tape._par_info`